gcs/constraints/smart_table.cc

```diff
@@ -0,0 +1,642 @@
+                        WeightedPseudoBooleanTerm minus_t{-1_i, smart_entry_flags[binary_entry_data]};
+                        entry_flags_sum.emplace_back(t);
+                        entry_flags_neg_sum.emplace_back(minus_t);
+                    },
+                    [&](const UnarySetEntry & unary_set_entry) {
+                        auto var = unary_set_entry.var;
+                        WeightedPseudoBooleanTerms set_value_sum{};
+                        WeightedPseudoBooleanTerms neg_set_value_sum{};
+                        for (const auto & val : unary_set_entry.values) {
+                            WeightedPseudoBooleanTerm t{1_i, var == val};
+                            WeightedPseudoBooleanTerm minus_t{-1_i, var == val};
+                            set_value_sum.emplace_back(t);
+                            neg_set_value_sum.emplace_back(minus_t);
+                        }
+
+                        auto flag = unary_set_entry.constraint_type == IN ? propagators.create_proof_flag("inset") : propagators.create_proof_flag("notinset");
+
+                        propagators.define_pseudoboolean_ge(initial_state, move(set_value_sum), 1_i,
+                            unary_set_entry.constraint_type == IN ? flag : ! flag);
+                        propagators.define_pseudoboolean_ge(initial_state, move(neg_set_value_sum), 0_i,
+                            unary_set_entry.constraint_type == IN ? ! flag : flag);
+
+                        WeightedPseudoBooleanTerm t{1_i, flag};
+                        WeightedPseudoBooleanTerm minus_t{-1_i, flag};
+                        entry_flags_sum.emplace_back(t);
+                        entry_flags_neg_sum.emplace_back(minus_t);
+                    },
+                    [&](UnaryValueEntry unary_value_entry) {
+                        Literal l = literal_from_unary_entry(unary_value_entry);
+                        WeightedPseudoBooleanTerm t{1_i, l};
+                        WeightedPseudoBooleanTerm minus_t{-1_i, l};
+                        entry_flags_sum.emplace_back(t);
+                        entry_flags_neg_sum.emplace_back(minus_t);
+                    }}
+                    .visit(entry);
+            }
+            auto tuple_len = Integer{static_cast<long long>(entry_flags_sum.size())};
+            propagators.define_pseudoboolean_ge(initial_state, move(entry_flags_sum), tuple_len, pb_selectors[tuple_idx]);
+            propagators.define_pseudoboolean_ge(initial_state, move(entry_flags_neg_sum), -tuple_len + 1_i, ! pb_selectors[tuple_idx]);
+        }
+    }
+
+    // Trigger when any var changes? Is this overkill?
+    Triggers triggers;
+    triggers.on_change = {_vars.begin(), _vars.end()};
+
+    vector<Forest> forests = build_forests(_tuples);
+
+    propagators.install(
+        [selectors, vars = _vars, tuples = _tuples, forests = forests](State & state) -> pair<Inference, PropagatorState> {
```

Comment:
You should be able to `move(_tuples)` and forests here.

Reply:
Fixed in c62e8914eb9e01df2a147f01713ff505d7e905e0